### PR TITLE
feat(tracers): Support Tracing Transactions and Blocks

### PR DESCRIPTION
**Pull request by calbera**
Requires refactor of plugin architecture

### CODE REVIEW

**Comment by itsdevbear** [comment]
Should we add debug_ tests to make sure it's good before we merge?

File: cosmos/x/evm/plugins/configuration/plugin.go [hunk 1]
```diff
@@ -33,22 +33,25 @@ import (
...
 // Plugin is the interface that must be implemented by the plugin.
 type Plugin interface {
 	plugins.BaseCosmosPolaris
 	core.ConfigurationPlugin
-	SetParams(params *types.Params)
-	GetParams() *types.Params
-	GetEvmDenom() string
+	SetParams(ctx sdk.Context, params *types.Params)
+	GetParams(ctx sdk.Context) *types.Params
+	GetEvmDenom(ctx context.Context) string
```
[the remark this answers is not in the document]
Comment (calbera) [reply cosmos/x/evm/plugins/configuration/plugin.go:45]: this one sometimes gets called from the RPC with a Context interface

[the remark this answers is not in the document]
Comment (itsdevbear) [reply cosmos/x/evm/plugins/configuration/plugin.go:45]: That's cosmos specific tho for sure

File: cosmos/x/evm/plugins/gas/plugin.go [hunk 2]
```diff
@@ -50,22 +51,19 @@ type plugin struct {
...
-}
-
-// Prepare implements the core.GasPlugin interface.
-func (p *plugin) Prepare(ctx context.Context) {
-	sCtx := sdk.UnwrapSDKContext(ctx)
-	p.gasMeter = sCtx.GasMeter()
-	p.blockGasMeter = sCtx.BlockGasMeter()
-}
+func NewPlugin(ctx context.Context) Plugin {
+	if !utils.Implements[sdk.Context](ctx) {
```
[the remark this answers is not in the document]
Comment (calbera) [reply cosmos/x/evm/plugins/gas/plugin.go:55]: yeah its just dealing with the context that now can come in from sdk __or__ from the rpc functions

[the remark this answers is not in the document]
Comment (itsdevbear) [reply cosmos/x/evm/plugins/gas/plugin.go:55]: Oh this is gonna cause fuckery, we should map this out

**Comment by itsdevbear** [comment]
We can probably kill this branch

**Pull request closed without merging**